src/main/java/decodes/cwms/CwmsSqlDatabaseIO.java

```diff
@@ -304,4 +318,98 @@ public IntervalDAI makeIntervalDAO()
 		return new CwmsIntervalDAO(this, dbOfficeId);
 	}
 
+	@Override
+	public Connection getConnection()
+	{
+		// Called from DAOs and DbIo to get a new connection from the pool.
+		if (conInfo == null || conInfo.getLoginInfo() == null)
+		{
+			Logger.instance().failure(module + ".getConnection -- loginInfo is null! DB not initialized?");
+			return null;
+		}
+		
+		Connection ret = null;
+		try
+		{
+			ret = CwmsDbConnectionPool.getInstance().getConnection(conInfo.getLoginInfo());
```

Comment:
I recommend adding the second String parameter in order to setup the end-to-end metrics.

Reply:
Done.